controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -115,6 +119,49 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return reconcile.Result{}, err
 	}
 
+	// Cleanup old namespace's resource, but keep old namespace
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	s, _ := labels.Parse("opendatahub.io/generated-namespace=true")
+	ODHNamespaceList, err := clientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: s.String(),
+	})
+	if !apierrs.IsNotFound(err) {
+		return reconcile.Result{}, err
+	}
+
+	resourceTypes := []client.Object{
+		&appsv1.Deployment{},
+		&corev1.Service{},
+		&corev1.Secret{},
+		&corev1.ConfigMap{},
+		&netv1.NetworkPolicy{},
+		&authv1.Role{},
+		&authv1.RoleBinding{},
+		&authv1.ClusterRole{},
+		&authv1.ClusterRoleBinding{},
+	}
+
+	for _, matchedNamespace := range ODHNamespaceList.Items {
+		if matchedNamespace.Name != namespace && matchedNamespace.Name != instance.Spec.Monitoring.Namespace {
+			for _, obj := range resourceTypes {
+				if err = r.Client.DeleteAllOf(ctx, obj, client.InNamespace(matchedNamespace.Name), client.MatchingLabels{"opendatahub.io/component": "true"}); err != nil {
```

Comment:
for this approach, it will require longer  downtime, right?
e.g DSCI first  create new namespace then change DSC then DSC take reconcile till it is done then user will be able to update in Yaml view.